README.md
# [Black Money React](https://demos.creative-tim.com/black-dashboard-react/#/dashboard) [![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&logo=twitter)](https://twitter.com/intent/tweet?url=https%3A%2F%2Fdemos.creative-tim.com%2Fblack-dashboard-react%2F%23%2Fadmin%2Fdashboard&text=Black%20Dashboard%20React%20by%20Creative%20Tim%20%7C%20Free%20React%20Admin%20Template&original_referer=https%3A%2F%2Fdemos.creative-tim.com%2Fblack-dashboard-react%2F&via=creativetim&hashtags=react%2Ccreativetim%2Cdark%2Cdashboard%2Cbootstrap%2Creactstrap%2Creactjs)

![version](https://img.shields.io/badge/version-1.2.1-blue.svg) ![license](https://img.shields.io/badge/license-MIT-blue.svg) [![GitHub issues open](https://img.shields.io/github/issues/creativetimofficial/black-dashboard-react.svg?maxAge=2592000)]() [![GitHub issues closed](https://img.shields.io/github/issues-closed-raw/creativetimofficial/black-dashboard-react.svg?maxAge=2592000)]() [![Chat](https://img.shields.io/badge/chat-on%20discord-7289da.svg)](https://discord.gg/E4aHAQy)

![Product Gif](./github-assets/black-dashboard-react.gif)

Black Money React is a beautiful [Bootstrap 4](http://getbootstrap.com/), [Reacstrap](https://reactstrap.github.io/) and [React](https://reactjs.org/) ([create-react-app](https://github.com/facebook/create-react-app)) Admin Money with a huge number of components built to fit together and look amazing. If you are looking for a tool to manage and visualize data about your business, this dashboard is the thing for you. It combines colors that are easy on the eye, spacious cards, beautiful typography, and graphics.
Black Money React comes packed with all plugins that you might need inside a project and documentation on how to get started. It is light and easy to use, and also very powerful.

Black Money React features over 16 individual components, giving you the freedom of choosing and combining. This means that there are thousands of possible combinations. Money components can take variations in color, that you can easily modify using SASS files. You will save a lot of time going from prototyping to full-functional code because all elements are implemented.
We thought about everything, so this dashboard comes with 2 versions, Dark Mode and Light Mode.

Special thanks go for the owners of these plugins:

- [Perfect Scrollbar](http://utatti.github.io/perfect-scrollbar/) for the nice scrollbars for windows.
- [React Chartjs](http://jerairrest.github.io/react-chartjs-2/) for the wonderful wrapper for charts.
- [Chartjs](https://www.chartjs.org/) for the wonderful charts.

We are very excited to share this dashboard with you and we look forward to hearing your feedback!

**Example Pages** We wanted to fully display the power of this dashboard, so the kit comes packed with examples showing you how to use the components.

## Table of Contents

- [Versions](#versions)
- [Demo](#demo)
- [Quick Start](#quick-start)
- [Documentation](#documentation)
- [File Structure](#file-structure)
- [Browser Support](#browser-support)
- [Resources](#resources)
- [Reporting Issues](#reporting-issues)
- [Technical Support or Questions](#technical-support-or-questions)
- [Licensing](#licensing)
- [Useful Links](#useful-links)

## Versions

[<img src="./github-assets/html.png" width="60" height="60" />](https://www.creative-tim.com/product/black-dashboard)
[<img src="./github-assets/react.svg" width="60" height="60" />](https://www.creative-tim.com/product/black-dashboard-react)[<img src="./github-assets/vuejs.png" width="60" height="60" />](https://www.creative-tim.com/product/vue-black-dashboard)

| HTML                                                                                                                                                                | React                                                                                                                                                                   | Vue                                                                                                                                                                        |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| [![Black Money HTML](https://s3.amazonaws.com/creativetim_bucket/products/93/thumb/opt_bd_thumbnail.jpg)](https://www.creative-tim.com/product/black-dashboard) | [![Black Money React](https://s3.amazonaws.com/creativetim_bucket/products/136/thumb/opt_bd_react.jpg)](https://www.creative-tim.com/product/black-dashboard-react) | [![Vue Black Money](https://s3.amazonaws.com/creativetim_bucket/products/99/thumb/opt_bd_vue_thumbnail.jpg)](https://www.creative-tim.com/product/vue-black-dashboard) |

## Demo

| Money                                                                                                             | User Profile                                                                                                            | Tables                                                                                                               | Maps                                                                                                       | Notification                                                                                                                         |
| --------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
| [![Start page](./github-assets/dashboard-page.png)](https://demos.creative-tim.com/black-dashboard-react/#/dashboard) | [![User profile page](./github-assets/user-page.png)](https://demos.creative-tim.com/black-dashboard-react/#/user-page) | [![Tables page ](./github-assets/table-page.png)](https://demos.creative-tim.com/black-dashboard-react/#/table-list) | [![Maps Page](./github-assets/maps-page.png)](https://demos.creative-tim.com/black-dashboard-react/#/maps) | [![Notification page](./github-assets/notifications-page.png)](https://demos.creative-tim.com/black-dashboard-react/#/notifications) |

[View More](https://demos.creative-tim.com/black-dashboard-react/#/dashboard).

## Quick start

Quick start options:

- Clone the repo: `git clone https://github.com/creativetimofficial/black-dashboard-react.git`.
- [Download from Github](https://github.com/creativetimofficial/black-dashboard-react/archive/master.zip).
- [Download from Creative Tim](https://www.creative-tim.com/product/black-dashboard-react).

## Documentation

The documentation for the Black Money React is hosted at our [website](https://demos.creative-tim.com/black-dashboard-react/#/documentation/tutorial).

## File Structure

Within the download you'll find the following directories and files:

```
black-dashboard-react
.
├── package.json
├── CHANGELOG.md
├── ISSUE_TEMPLATE.md
├── LICENSE.md
├── README.md
├── Documentation
│   └── documentation.html
├── github-assets
│   └── react.svg
├── public
│   ├── index.html
│   └── manifest.json
└── src
    ├── index.js
    ├── logo-white.svg
    ├── logo.svg
    ├── routes.js
    ├── assets
    │   ├── css
    │   ├── demo
    │   ├── fonts
    │   ├── img
    │   └── scss
    │       ├── black-dashboard
    │       │   ├── bootstrap
    │       │   │   ├── mixins
    │       │   │   └── utilities
    │       │   ├── custom
    │       │   │   ├── cards
    │       │   │   ├── mixins
    │       │   │   ├── utilities
    │       │   │   └── vendor
    │       └── black-dashboard.scss
    ├── components
    │   ├── FixedPlugin
    │   │   └── FixedPlugin.js
    │   ├── Footer
    │   │   └── Footer.js
    │   ├── Navbars
    │   │   ├── AdminNavbar.js
    │   │   └── RTLNavbar.js
    │   └── Sidebar
    │       └── Sidebar.js
    ├── layouts
    │   ├── Admin
    │   │   └── Admin.js
    │   └── RTL
    │       └── RTL.js
    ├── variables
    │   └── charts.js
    └── views
        ├── Money.js
        ├── Icons.js
        ├── Map.js
        ├── Notifications.js
        ├── Rtl.js
        ├── TableList.js
        ├── Typography.js
        └── UserProfile.js
```

## Browser Support

At present, we officially aim to support the last two versions of the following browsers:

<img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/chrome.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/firefox.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/edge.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/safari.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/opera.png" width="64" height="64">

## Resources

- Demo: https://demos.creative-tim.com/black-dashboard-react/#/dashboard
- Download Page: https://www.creative-tim.com/product/black-dashboard-react
- Documentation: https://demos.creative-tim.com/black-dashboard-react/#/documentation/tutorial
- License Agreement: https://www.creative-tim.com/license
- Support: https://www.creative-tim.com/contact-us
- Issues: [Github Issues Page](https://github.com/creativetimofficial/black-dashboard-react/issues)

## Reporting Issues

We use GitHub Issues as the official bug tracker for the Black Money React. Here are some advices for our users that want to report an issue:

1. Make sure that you are using the latest version of the Black Money React. Check the CHANGELOG from your dashboard on our [website](https://www.creative-tim.com/).
2. Providing us reproducible steps for the issue will shorten the time it takes for it to be fixed.
3. Some issues may be browser specific, so specifying in what browser you encountered the issue might help.

## Technical Support or Questions

If you have questions or need help integrating the product please [contact us](https://www.creative-tim.com/contact-us) instead of opening an issue.

## Licensing

- Copyright 2022 Creative Tim (https://www.creative-tim.com)
- Licensed under MIT (https://github.com/creativetimofficial/black-dashboard-react/blob/master/LICENSE.md)

## Useful Links

More products from Creative Tim: <https://www.creative-tim.com/products>

Tutorials: <https://www.youtube.com/channel/UCVyTG4sCw-rOvB9oHkzZD1w>

Freebies: <https://www.creative-tim.com/products>

Affiliate Program (earn money): <https://www.creative-tim.com/affiliates/new>

Social Media:

Twitter: <https://twitter.com/CreativeTim>

Facebook: <https://www.facebook.com/CreativeTim>

Dribbble: <https://dribbble.com/creativetim>

Google+: <https://plus.google.com/+CreativetimPage>

Instagram: <https://instagram.com/creativetimofficial>


CHANGELOG.md
# CHANGELOG

## [1.2.1] 2022-06-29

- Update dependencies
- Migrate from node-sass to sass

## [1.2.0] 2020-12-11

### Bug fixing

- Added hooks support
- https://github.com/creativetimofficial/black-dashboard-react/issues/19
- https://github.com/creativetimofficial/black-dashboard-react/issues/17

### Major style changes

- Money Bootstrap imports from `src/assets/scss/black-dashboard-react/bootstrap` have been changed with imports from `node_modules` from `Bootstrap`

### Deleted components

### Added components

### Deleted dependencies

- history (instead of using `history`, we'll be using `BrowserRouter` from `react-router-dom`)
- @types/googlemaps
- @types/react
- @types/markerclustererplus
- react-google-maps (Instead of it, we're going to use simple plain JS Google maps)

### Added dependencies

- node-sass-package-importer@5.3.2 (for importing `Bootstrap` from `node_modules`)
- bootstrap@4.5.3 (instead of using downloaded zip of `Bootstrap`, we'll use it from `node_modules`)
- jquery@3.5.1 (to stop `Bootstrap` warning on a clean install)

### Updated dependencies

```
@fortawesome/fontawesome-free    5.13.0   →   5.15.1
chart.js                          2.9.3   →    2.9.4
node-sass                        4.13.1   →   4.14.1
react                           16.13.1   →   17.0.1
react-chartjs-2                   2.9.0   →   2.11.1
react-dom                       16.13.1   →   17.0.1
react-notification-alert         0.0.12   →   0.0.13
react-router-dom                  5.1.2   →    5.2.0
react-scripts                     3.4.1   →    4.0.1
reactstrap                        8.4.1   →    8.7.1
eslint-plugin-flowtype           3.13.0   →    5.2.0
typescript                        3.8.3   →    4.1.2
```

### Important Notes

**The jQuery and TypeScript dependencies are installed only to stop console warnings on install. They are not actually used in our product. So the product is not based on jQuery, and it is not based on TypeScript!**

### Warning

_Some warnings may appear when running the installation command, but they do not affect the UI or the functionality of the product._
_The following warnings will appear when running the installation command, but they do not affect the UI or the functionality of the product (they will be solved in our next update - they come from the plugins that we are using, and they haven't yet upgraded to the latest React version):_

```
npm WARN react-popper@1.3.7 requires a peer of react@0.14.x || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN create-react-context@0.3.0 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
```

Happy Hacking!

## [1.1.0] 2020-03-13

### Bug fixing

- Github issues:
  - https://github.com/creativetimofficial/black-dashboard-react/issues/8
  - https://github.com/creativetimofficial/black-dashboard-react/issues/7
  - https://github.com/creativetimofficial/black-dashboard-react/issues/6 (not really an issue, but was old and needed to close it)
  - https://github.com/creativetimofficial/black-dashboard-react/issues/4

### Major style changes

### Deleted components

### Added components

### Deleted dependencies

- bootstrap (styles are already added inside the scss folder)

### Added dependencies

- To stop consolde warnings on installation:
  - eslint-plugin-flowtype@3.13.0
  - @types/markerclustererplus@2.1.33
  - @types/googlemaps@3.39.3
  - typescript@3.8.3
- For Creative Tim copyrights
  - gulp@4.0.2
  - gulp-append-prepend@1.0.8

### Updated dependencies

```
@types/react               16.4.16   →   16.9.23
chart.js                    ^2.7.2   →     2.9.3
history                      4.7.2   →    4.10.1
node-sass                    4.9.3   →    4.13.1
perfect-scrollbar            1.4.0   →     1.5.0
prop-types                 ^15.6.2   →    15.7.2
react                       16.5.2   →   16.13.0
react-chartjs-2             ^2.7.4   →     2.9.0
react-dom                   16.5.2   →   16.13.0
react-notification-alert     0.0.8   →    0.0.12
react-router-dom             4.3.1   →     5.1.2
react-scripts                2.0.4   →     3.4.0
reactstrap                   6.5.0   →     8.4.1
```

### Warning

**There are some errors that come from the react-google-maps components, since they are using old React v16 syntax, this, at the moment does not affect the UI or the functionality of the product. If the issue will perssit in React v17, we will drop the usage of these components and replace them with other maps components. The following warnings on installation could not be solved, due to some of our dependencies, however, they do not affect the functionality or the UI of the product:**

```
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated popper.js@1.16.1: Popper changed home, find its new releases at @popperjs/core
```

**Also, the above, are only present in development, and not in production.**

## [1.0.0] 2018-11-29

### Original Release

- Added Reactstrap as base framework
- Added design from PRODUCT NAME by Creative Tim


src/views/Money.js
import {useState, useEffect} from "react";
// nodejs library that concatenates classes
import classNames from "classnames";
// react plugin used to create charts
import { Line, Bar } from "react-chartjs-2";

// reactstrap components
import {
  Button,
  ButtonGroup,
  Card,
  CardHeader,
  CardBody,
  CardTitle,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  UncontrolledDropdown,
  Label,
  FormGroup,
  Input,
  Table,
  Row,
  Col,
  UncontrolledTooltip
} from "reactstrap";

// core components
import {
  totalSpendingChart,
  artBoughtChart,
  chartExample3,
  chartExample4
} from "variables/charts.js";



function Money(props) {

  const fakeData = {
    dataAnnual: [
      {
        id: 1,
        title: 'Virginia Maria',
        artistName: 'Arriba Donatelo',
        price: 10300,
        date:"16/05/2019"
      },
      {
        id: 13,
        title: 'Happiness',
        artistName: 'John Dofferman',
        price: 1000,
        date:"06/05/2019"
      },
      {
        id: 14,
        title: 'Maze',
        artistName: 'Maria Antonieta',
        price: 50500,
        date:"06/05/2020"
      },
      {
        id: 15,
        title: 'Saint Peter',
        artistName: 'Bonitchelo',
        price: 22500,
        date:"06/05/2020"
      },
      {
        id: 17,
        title: 'Grand Amusement',
        artistName: 'Epicacio D. Antonio',
        price: 20000,
        date:"14/11/2020"
      },
      {
        id: 18,
        title: 'Orgy with the Trans',
        artistName: 'The Killer Mamonas',
        price: 3630,
        date:"06/05/2021"
      },
      {
        id: 19,
        title: 'Saint Peter',
        artistName: 'Bonitchelo',
        price: 3630,
        date:"06/05/2021"
      },
      {
        id: 20,
        title: 'Saint Peter',
        artistName: 'Bonitchelo',
        price: 3630,
        date:"06/05/2022"
      },
      {
        id: 2,
        title: 'Operários',
        artistName: 'Tarsila do Amaral',
        price: 600,
        date:"19/01/2022"
      },
      {
        id: 3,
        title: 'Blue in Rio',
        artistName: 'Pixarlo',
        price: 520,
        date:"20/01/2022"
      },
      {
        id: 4,
        title: 'Canva and Cry',
        artistName: 'Eduardo Pinoquio',
        price: 2420,
        date:"12/02/2022"
      },
      {
        id: 5,
        title: 'Mama sua',
        artistName: 'Frederique D\'Art',
        price: 700,
        date:"09/03/2022"
      },
      {
        id: 6,
        title: 'Mada Star',
        artistName: 'Bonitchelo Guapo',
        price: 900,
        date:"17/04/2022"
      },
      {
        id: 7,
        title: 'Modus Operandis',
        artistName: 'Jorge de Souza',
        price: 750,
        date:"02/05/2022"
      },
      {
        id: 8,
        title: 'Maria em Braços',
        artistName: 'Santo Antonio',
        price: 450,
        date:"15/05/2022"
      },
      {
        id: 9,
        title: 'Santo Demonio',
        artistName: 'Chaves Companhia',
        price: 2200,
        date:"29/06/2022"
      },
      {
        id: 10,
        title: 'Vivendo Adoidado',
        artistName: 'Feris Bouler',
        price: 1014,
        date:"12/07/2022"
      },
      {
        id: 11,
        title: 'In your arms only',
        artistName: 'Eduran Sheran',
        price: 1110,
        date:"22/07/2022"
      },
      {
        id: 12,
        title: 'Saint Peter',
        artistName: 'Bonitchelo',
        price: 3630,
        date:"06/05/2022"
      },
    ],
    dataMonth: [
        {
          id: 2,
          title: 'Operários',
          artistName: 'Tarsila do Amaral',
          price: 600,
          date:"19/01/2022"
        },
        {
          id: 3,
          title: 'Blue in Rio',
          artistName: 'Pixarlo',
          price: 520,
          date:"20/01/2022"
        },
        {
          id: 4,
          title: 'Canva and Cry',
          artistName: 'Eduardo Pinoquio',
          price: 2420,
          date:"12/02/2022"
        },
        {
          id: 5,
          title: 'Mama sua',
          artistName: 'Frederique D\'Art',
          price: 700,
          date:"09/03/2022"
        },
        {
          id: 6,
          title: 'Mada Star',
          artistName: 'Bonitchelo Guapo',
          price: 900,
          date:"17/04/2022"
        },
        {
          id: 7,
          title: 'Modus Operandis',
          artistName: 'Jorge de Souza',
          price: 750,
          date:"02/05/2022"
        },
        {
          id: 8,
          title: 'Maria em Braços',
          artistName: 'Santo Antonio',
          price: 450,
          date:"15/05/2022"
        },
        {
          id: 9,
          title: 'Santo Demonio',
          artistName: 'Chaves Companhia',
          price: 2200,
          date:"29/06/2022"
        },
        {
          id: 10,
          title: 'Vivendo Adoidado',
          artistName: 'Feris Bouler',
          price: 1014,
          date:"12/07/2022"
        },
        {
          id: 11,
          title: 'In your arms only',
          artistName: 'Eduran Sheran',
          price: 1110,
          date:"22/07/2022"
        },
        {
          id: 12,
          title: 'Saint Peter',
          artistName: 'Bonitchelo',
          price: 3630,
          date:"06/05/2022"
        },
    ]
  }

  const [loadingSpending, setLoadingSpending] = useState(true);
  const [selectedSpendingChart, setSelectedSpendingChart] = useState("annual");
  const [spendingAnnualGraph, setSpendingDataGraph] = useState();
  const [spendingMonthlyGraph, setSpendingMonthlyDataGraph] = useState();


  const [loadingBought, setLoadingBought] = useState(false);
  const [selectedBoughtChart, setSelectedBoughtChart] = useState("monthlyBoughtArt");
  const [artBoughtData, setArtBoughtData] = useState();

  useEffect(() => {
    fetch("http://localhost:8080/permanent-collection/spent-per-date/year", {
      method: "GET",
      mode: "cors",
    })
      .then(res => res.json())
      .then(
        async (result) => {
          const labels = [];
          const data = [];
          result.forEach( (result) => {
            labels.push(result.date)
            data.push(result.quantity)
          })
          setSpendingDataGraph({
              labels: labels,
              datasets: [
                {
                  label: "Something",
                  fill: true,
                  borderColor: "#1f8ef1",
                  borderWidth: 2,
                  borderDash: [],
                  borderDashOffset: 0.0,
                  pointBackgroundColor: "#1f8ef1",
                  pointBorderColor: "rgba(255,255,255,0)",
                  pointHoverBackgroundColor: "#1f8ef1",
                  pointBorderWidth: 20,
                  pointHoverRadius: 4,
                  pointHoverBorderWidth: 15,
                  pointRadius: 4,
                  data: data
                }
              ]
            }
          );
          setLoadingSpending(false);
        },
        (error) => {
          setLoadingSpending(false);
          console.log(error);
        }
      )
  }, [])

  useEffect(() => {
    fetch("http://localhost:8080/permanent-collection/spent-per-date/month", {
      method: "GET",
      mode: "cors",
    })
      .then(res => res.json())
      .then(
        async (result) => {
          const labels = [];
          const data = [];
          result.forEach( (result) => {
            labels.push(result.date)
            data.push(result.quantity)
          })
          setSpendingMonthlyDataGraph({
              labels: labels,
              datasets: [
                {
                  label: "Something",
                  fill: true,
                  borderColor: "#1f8ef1",
                  borderWidth: 2,
                  borderDash: [],
                  borderDashOffset: 0.0,
                  pointBackgroundColor: "#1f8ef1",
                  pointBorderColor: "rgba(255,255,255,0)",
                  pointHoverBackgroundColor: "#1f8ef1",
                  pointBorderWidth: 20,
                  pointHoverRadius: 4,
                  pointHoverBorderWidth: 15,
                  pointRadius: 1,
                  data: data
                }
              ]
            }
          );
          setLoadingSpending(false);
        },
        (error) => {
          setLoadingSpending(false);
          console.log(error);
        }
      )
  }, [])


  useEffect(() => {
    fetch("http://localhost:8080/permanent-collection/objects")
      .then(res => res.json())
      .then(
        (result) => {
          console.log(result)
          setLoadingBought(false);
          setArtBoughtData(result);
        },
        (error) => {
          setLoadingSpending(false);
          console.log(error);
        }
      )
  }, [])

  return (
    <>
      <div className="content">
        <Row>
          <Col xs="12">
            <Card className="card-chart">
              <CardHeader>
                <Row>
                  <Col className="text-left" sm="6">
                    <h5 className="card-category">Todos Objetos de Arte</h5>
                    <CardTitle tag="h2">Total Gasto</CardTitle>
                  </Col>
                  <Col sm="6">
                    <ButtonGroup
                      className="btn-group-toggle float-right"
                      data-toggle="buttons"
                    >
                      <Button
                        tag="label"
                        className={classNames("btn-simple", {
                          active: selectedSpendingChart === "monthly"
                        })}
                        color="info"
                        id="monthlySpent"
                        size="sm"
                        onClick={() => setSelectedSpendingChart("monthly")}
                      >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Mensal
                        </span>
                        <span className="d-block d-sm-none">
                          <i className="tim-icons icon-single-02" />
                        </span>
                      </Button>
                      <Button
                        color="info"
                        id="annualSpent"
                        size="sm"
                        tag="label"
                        className={classNames("btn-simple", {
                          active: selectedSpendingChart === "annual"
                        })}
                        onClick={() => setSelectedSpendingChart("annual")}
                      >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Anual
                        </span>
                        <span className="d-block d-sm-none">
                          <i className="tim-icons icon-gift-2" />
                        </span>
                      </Button>
                    </ButtonGroup>
                  </Col>
                </Row>
              </CardHeader>
              <CardBody>
                <div className="chart-area">
                  {!loadingSpending && selectedSpendingChart==="annual"
                    ? (
                      <Line
                        data={spendingAnnualGraph}
                        options={totalSpendingChart.options}
                      />
                    )
                    : (
                      <Line
                        data={spendingMonthlyGraph}
                        options={totalSpendingChart.options}
                      />
                    )
                  }
                </div>
              </CardBody>
            </Card>
          </Col>
        </Row>
        <Row>
          <Col >
            <Card>
              <CardHeader>
                <CardTitle tag="h4">Compras de Objetos</CardTitle>
              </CardHeader>
              <Col sm="6">
                <ButtonGroup
                  className="btn-group-toggle float-right"
                  data-toggle="buttons"
                >
                  <Button
                    tag="label"
                    className={classNames("btn-simple", {
                      active: selectedBoughtChart === "monthlyBoughtArt"
                    })}
                    color="info"
                    id="boughtMonthly"
                    size="sm"
                    onClick={() => {
                      setArtBoughtData("dataMonth")
                      setSelectedBoughtChart("monthlyBoughtArt")
                    }}
                  >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Mensal
                        </span>
                    <span className="d-block d-sm-none">
                          <i className="tim-icons icon-single-02" />
                        </span>
                  </Button>
                  <Button
                    color="info"
                    id="boughtAnnual"
                    size="sm"
                    tag="label"
                    className={classNames("btn-simple", {
                      active: selectedBoughtChart === "annualBoughtArt"
                    })}
                    onClick={() => {
                      setArtBoughtData("dataAnnual")
                      setSelectedBoughtChart("annualBoughtArt")
                    }}
                  >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Anual
                        </span>
                    <span className="d-block d-sm-none">
                          <i className="tim-icons icon-gift-2" />
                        </span>
                  </Button>
                </ButtonGroup>
              </Col>
              <CardBody>
                <div className={"divMeu"}>
                <Table responsive>
                  <thead className="text-primary">
                  <tr>
                    <th>Titulo da Obra</th>
                    <th>Nome do Artista</th>
                    <th>Preço</th>
                    <th className="text-center">Date</th>

                  </tr>
                  </thead>
                  <tbody>
                  { artBoughtData
                    ? artBoughtData.map((eachItem) => {
                      return(
                        <tr>
                          <td>{eachItem.title}</td>
                          <td>{eachItem.artistName}</td>
                          <td className="text-center">${eachItem.price}</td>
                          <td>{eachItem.date}</td>
                        </tr>
                      )
                    })
                    : null}
                  </tbody>
                </Table>
                </div>
              </CardBody>
            </Card>
          </Col>
        </Row>
      </div>
    </>
  );
}

export default Money;
